src/tree/split_evaluator.cc

```diff
@@ -175,22 +146,6 @@ XGBOOST_REGISTER_SPLIT_EVALUATOR(ElasticNet, "elastic_net")
     return new ElasticNet(std::move(inner));
   });
 
-/*! \brief Encapsulates the parameters required by the MonotonicConstraint
-        split evaluator
-*/
-struct MonotonicConstraintParams
-    : public XGBoostParameter<MonotonicConstraintParams> {
-  std::vector<bst_int> monotone_constraints;
```

Comment:
Is this field now part of TrainParam?

Reply:
Indeed it is: https://github.com/dmlc/xgboost/blob/df9bdbbcb9713389759293a67e225b0985a52346/src/tree/param.h#L75

Reply:
@hcho3 Yes: https://github.com/dmlc/xgboost/blob/2dcb62ddfb2e10d58358195992df829e6d558c83/src/tree/param.h#L75

I think the one in TrainParam is older than the one in split evaluator.

Reply:
Nice job deduplicating this parameter. Having one source of parameters will make our life easier.